### PR TITLE
Add fancy stream migration patterns and stream-related best-practice

### PR DESCRIPTION
**Pull request by drandarov-io**
Pull-Request over GitKraken ;)

### CODE REVIEW

File: src/test/java/de/stevenschwenke/java/java8workshop/C_13_Fancy_Stream_Migration_Patterns.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+     * Just as simple as the one above, this is basically a contains check that incorportates some sort of field of the
+     * pojo so a simple {@link java.util.List#contains(Object)} won't do it. There are also other kinds of data models
+     * that don't include any kind of containment-check.
+     */
+    @Test
+    public void anyMatchPattern() {
+        Boolean neueMarkeGefunden = false;
+        for(DummyFruit DummyFruit : dummyFruits.values()) {
+            if("Baby Banana".equals(DummyFruit.getName()))
+                neueMarkeGefunden = true;
```
Comment (stevenschwenke) [inline src/test/java/de/stevenschwenke/java/java8workshop/C_13_Fancy_Stream_Migration_Patterns.java]: Could you please refactor that? :)

File: src/test/java/de/stevenschwenke/java/java8workshop/C_13_Fancy_Stream_Migration_Patterns.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        /*
+            | | | | | | |
+            v v v v v v v
+        */
+
+        //Better and sexier code
+        assertTrue(dummyFruits.values().stream().anyMatch(DummyFruit -> "Baby Banana".equals(DummyFruit.getName())));
+    }
+
+    //More Patterns to come
```
Comment (stevenschwenke) [inline src/test/java/de/stevenschwenke/java/java8workshop/C_13_Fancy_Stream_Migration_Patterns.java]: Hope so, the patterns above are great! However, until there are new more awesome sexy patterns, could you remove this hidden TODO?

**Review by stevenschwenke — CHANGES_REQUESTED** [review]
Thanks for your PR, it's great! However, could you please fix this two minor issues? Thanks!

File: src/test/java/de/stevenschwenke/java/java8workshop/C_13_Fancy_Stream_Migration_Patterns.java (hunk 1 above)
Comment (drandarov-io) [reply src/test/java/de/stevenschwenke/java/java8workshop/C_13_Fancy_Stream_Migration_Patterns.java]: Assuming you meant either the missing braces or the Boolean-Object, I just fixed it. ;)

**Pull request merged**